src/main/scala/cromwell/webservice/CromwellApiHandler.scala

```diff
@@ -2,46 +2,70 @@ package cromwell.webservice
     Props(new CromwellApiHandler(workflowManagerActorRef))
   }
 
-  sealed trait WorkflowManagerMessage
-
-  final case class WorkflowSubmit(source: WorkflowSourceFiles) extends WorkflowManagerMessage
-  final case class WorkflowQuery(parameters: Seq[(String, String)]) extends WorkflowManagerMessage
-  final case class WorkflowStatus(id: WorkflowId) extends WorkflowManagerMessage
-  final case class WorkflowOutputs(id: WorkflowId) extends WorkflowManagerMessage
-  final case class WorkflowAbort(id: WorkflowId) extends WorkflowManagerMessage
-  final case class CallOutputs(id: WorkflowId, callFqn: String) extends WorkflowManagerMessage
-  final case class CallStdoutStderr(id: WorkflowId, callFqn: String) extends WorkflowManagerMessage
-  final case class WorkflowStdoutStderr(id: WorkflowId) extends WorkflowManagerMessage
-  final case class CallCaching(id: WorkflowId, parameters: QueryParameters, callName: Option[String]) extends WorkflowManagerMessage
-  final case class WorkflowMetadata(id: WorkflowId) extends WorkflowManagerMessage
+  sealed trait ApiHandlerMessage
+
+  final case class ApiHandlerWorkflowSubmit(source: WorkflowSourceFiles) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowQuery(parameters: Seq[(String, String)]) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowStatus(id: WorkflowId) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowOutputs(id: WorkflowId) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowAbort(id: WorkflowId) extends ApiHandlerMessage
+  final case class ApiHandlerCallOutputs(id: WorkflowId, callFqn: String) extends ApiHandlerMessage
+  final case class ApiHandlerCallStdoutStderr(id: WorkflowId, callFqn: String) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowStdoutStderr(id: WorkflowId) extends ApiHandlerMessage
+  final case class ApiHandlerCallCaching(id: WorkflowId, parameters: QueryParameters, callName: Option[String]) extends ApiHandlerMessage
+  final case class ApiHandlerWorkflowMetadata(id: WorkflowId) extends ApiHandlerMessage
+
+  sealed trait WorkflowManagerResponse
+
+  sealed trait WorkflowManagerSuccessResponse extends WorkflowManagerResponse
+
+  sealed trait WorkflowManagerFailureResponse extends WorkflowManagerResponse {
+    def failure: Throwable
+  }
+
+  final case class WorkflowManagerSubmitSuccess(id: WorkflowId) extends WorkflowManagerSuccessResponse
+  final case class WorkflowManagerSubmitFailure(override val failure: Throwable) extends WorkflowManagerFailureResponse
+  final case class WorkflowManagerWorkflowOutputsSuccess(id: WorkflowId, outputs: engine.WorkflowOutputs) extends WorkflowManagerSuccessResponse
+  final case class WorkflowManagerWorkflowOutputsFailure(id: WorkflowId, override val failure: Throwable) extends WorkflowManagerFailureResponse
+  final case class WorkflowManagerStatusSuccess(id: WorkflowId, state: Option[WorkflowState]) extends WorkflowManagerSuccessResponse
```

Comment:
Actually this is a different problem: a "success" with a `None` status isn't legit, that means the specified workflow ID doesn't exist.  So really that scenario should be a failure and the success message shouldn't wrap the status in an `Option`.